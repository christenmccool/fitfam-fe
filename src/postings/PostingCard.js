import React from 'react';
import { NavLink as RouterLink } from 'react-router-dom';

import Card from '@mui/material/Card';
import CardActionArea from '@mui/material/CardActionArea';
import CardContent from '@mui/material/CardContent';
import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';

import PostingEditBar from '../postings/PostingEditBar';
import ResultDashboard from '../results/ResultDashboard';

/** Summary information about a posting
 * Includes ResultDashboard 
 *
 * PostingList -> PostingCardList -> PostingCard -> ResultDashboard
 * Card links to PostingDetail
 */
const PostingCard = ({ id, woName, woDescription, maxHeight, isUserPost }) => {

  return (
    <Card 
      variant="outlined" 
      align="center" 
      sx={{maxHeight, p:1}}
    >
      <CardContent>
          <Box p={2}>
            <CardActionArea 
              component={RouterLink} 
              to={`/postings/${id}`}
            > 
              <Typography 
                variant="h4" 
                color="secondary" 
                gutterBottom
              >
                {woName}
              </Typography>
              <Typography 
                variant={woName ? "h6" : "h4"} 
                style={{whiteSpace: "pre-wrap", wordWrap: "break-word"}}
              >
                {woDescription}
              </Typography>
            </CardActionArea>
            
            {isUserPost ?
              <PostingEditBar
                postId={id}
              />
              : null
            }
          </Box>

          <ResultDashboard 
            postId={id}
          />
      </CardContent>
    </Card>
  )
}

export default PostingCard;
