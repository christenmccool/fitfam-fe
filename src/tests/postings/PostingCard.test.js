import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import PostingCard from "../../postings/PostingCard";


it('renders without crashing', function () {
  render(
    <MemoryRouter>
      <PostingCard 
        id={1}
        woName={"Test Workout Name 1"} 
        woDescription={"Do this hard thing"}
        startExpanded={false}
        postByUser={false}
        isUserWo={false}
        isUserPosting={false}
      />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <PostingCard 
        id={1}
        woName={"Test Workout Name 1"} 
        woDescription={"Do this hard thing"}
        startExpanded={false}
        postByUser={false}
        isUserWo={false}
        isUserPosting={false}
      />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});


it("renders workout name but not description if not expanded", async function () {
  render(
    <MemoryRouter>
      <PostingCard 
        id={1}
        woName={"Test Workout Name 1"} 
        woDescription={"Do this hard thing"}
        startExpanded={false}
        postByUser={false}
        isUserWo={false}
        isUserPosting={false}
    />
    </MemoryRouter>
  );
  expect(screen.queryByText("Test Workout Name 1")).toBeInTheDocument();
  expect(screen.queryByText("Do this hard thing")).not.toBeInTheDocument();
});







