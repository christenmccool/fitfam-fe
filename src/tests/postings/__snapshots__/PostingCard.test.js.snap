// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    align="center"
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1l09cnc-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-566zuw-MuiCardContent-root"
    >
      <a
        class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
        href="/postings/1"
        tabindex="0"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-i0231a-MuiTypography-root"
        >
          Test Workout Name 1
        </h4>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
      <div
        class="MuiBox-root css-eldtq0"
      >
        <button
          aria-expanded="false"
          aria-label="show more"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-3ogxn6-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
